lib/unifex/code_generator/base_types/list.ex

```diff
@@ -11,6 +11,25 @@ defmodule Unifex.CodeGenerator.BaseTypes.List do
   alias Unifex.CodeGenerator.BaseType
 
   @impl BaseType
+  def generate_native_type(ctx)
+
+  def generate_native_type(%{mode: :const_if_not_ptr_on_ptr} = ctx) do
+    ptr? =
+      BaseType.generate_native_type(ctx.subtype, ctx.mode, ctx.generator, %{ctx | mode: :default})
+      |> String.replace(" ", "")
+      |> String.match?(~r"\*$")
+
+    mode = if not ptr?, do: :const, else: :default
+    ctx = %{ctx | mode: mode}
+
+    [
+      "#{BaseType.generate_native_type(ctx.subtype, ctx.mode, ctx.generator, ctx)} #{
+        if not ptr?, do: "const"
+      }*",
+      {"unsigned int", "_length"}
+    ]
```

Comment:
maybe we can just call generate_native_type recursively instead?

Reply:
If you mean, to generate `const` in native code in recursive call of `generate_native_type` - it is generated there too, but only if value of `ptr?` is set to `false`. Take a look at 3 lines of code above mentioned fragment